components/constellation/constellation.rs

```diff
@@ -1543,7 +1607,16 @@ where
                 self.handle_media_session_action_msg(action);
             },
             FromCompositorMsg::WebViewVisibilityChanged(webview_id, visible) => {
+                // TODO is this correct?
+                self.webviews.set_native_window_visibility(visible);
                 self.notify_webview_visibility(webview_id, visible);
+                let webviews = self.webviews.iter().map(|(&id, _)| id).collect::<Vec<_>>();
+                for top_level_browsing_context_id in webviews {
+                    let visible = self
+                        .webviews
+                        .is_effectively_visible(top_level_browsing_context_id);
+                    self.notify_webview_visibility(top_level_browsing_context_id, visible);
+                }
```

Comment:
I update the behavior of `RaiseWebViewToTop`, `ShowWebView`, `HideWebView`, and `WebViewVisibilityChanged`. Let me know if those are correct.
I haven't renamed it to `MarkWebViewInvisible` because I think there's still need a way to `mark_webview_no_invisible`.

Reply:
looks good, except for a few issues:

- HideWebView should mark_webview_not_shown, not mark_webview_invisible
- WebViewVisibilityChanged should not call mark_webview_shown or mark_webview_not_shown
- WebViewManager.shown_webviews doc comment should be “webviews that are being shown by the embedder, regardless of whether they are actually visible”
- WebViewManager.invisible_webviews doc comment should be “webviews that are known to be invisible, regardless of whether they are being shown”

Reply:
Updated in [`8083275`](https://github.com/servo/servo/pull/31417/commits/8083275411114606e24954af39e7876d287e89c6), and the variant is renamed in [`258d520`](https://github.com/servo/servo/pull/31417/commits/258d5202ea13ab2c2e2dee03a810c061f2e22b49)